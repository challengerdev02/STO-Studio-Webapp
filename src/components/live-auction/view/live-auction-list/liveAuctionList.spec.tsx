import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { LiveAuctionList } from './index';
import { format } from 'date-fns';

describe('LiveAuctionList component', function () {
  afterEach(cleanup);
  let props = {
    data: [
      {
        cryptoValue: '7.00698',
        cryptoCurrency: 'ETH',
        user: '0x658d22456633334234iF0cC',
        date: format(new Date('2021-11-17T01:00:00'), 'yyyy-MM-dd, HH:mm aaa'),
        userAvatar: '/verified-account-avatar.svg',
      },
      {
        cryptoValue: '7.00698',
        cryptoCurrency: 'ETH',
        user: '0x658d22456633334234iF0cC',
        date: format(new Date('2021-11-17T01:00:00'), 'yyyy-MM-dd, HH:mm aaa'),
        userAvatar: '/verified-account-avatar.svg',
      },
    ],
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(<LiveAuctionList {...props} {...extraProps} />);
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should check that the LiveAuctionList component displays', () => {
    const { getByTestId } = setup();

    expect(getByTestId('live-auction-list')).toBeInTheDocument();
  });

  it('should test the LiveAuctionList receives a crypto value props', () => {
    const { container } = setup();
    const crypto_value = container.querySelector('.ant-list-item-meta-title');
    expect(crypto_value).toBeInTheDocument();
    expect(crypto_value).toHaveTextContent('7.00698');
  });

  it('should test that the crypto value props the LiveAuctionList receives has a currency type ', () => {
    const { container } = setup();
    const currency_type = container.querySelector('.ant-list-item-meta-title');
    expect(currency_type).toBeInTheDocument();
    expect(currency_type).toHaveTextContent('ETH');
  });

  it('should test the LiveAuctionList receives a user blockchain address props', () => {
    const { container } = setup();

    const user = container.querySelector('.ant-list-item-meta-title');
    expect(user).toBeInTheDocument();
    expect(user).toHaveTextContent('7.00698 ETH by 0x658d22456633334...');
  });

  it('should test the LiveAuctionList receives a bid date props', () => {
    const { container } = setup();

    const date = container.querySelector('.ant-list-item-meta-description');
    expect(date).toBeInTheDocument();
    expect(date).toHaveTextContent('2021-11-17, 01:00 am');
  });
});
