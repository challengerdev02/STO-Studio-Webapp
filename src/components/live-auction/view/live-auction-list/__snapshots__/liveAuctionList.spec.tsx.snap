// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LiveAuctionList component should match with snapshot 1`] = `
<div>
  <div
    class="ant-list ant-list-split"
    data-testid="live-auction-list"
  >
    <div
      class="ant-spin-nested-loading"
    >
      <div
        class="ant-spin-container"
      >
        <ul
          class="ant-list-items"
        >
          <li
            class="ant-list-item"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-avatar"
              >
                <div
                  class="sc-bdvvtL kGscza"
                >
                  <div
                    class="sc-gsDKAQ eNKYiV"
                  >
                     
                    <img
                      alt=""
                      class="sc-dkPtRN jsPQsZ"
                      height="28px"
                      src="/verified-account-avatar.svg"
                      width="26px"
                    />
                  </div>
                </div>
              </div>
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  <span
                    class="sc-jRQBWg jfyBYj"
                  >
                    <span
                      data-testid="crypto_value"
                    >
                      7.00698
                    </span>
                     
                    <span
                      class="sc-hKwDye emujjc"
                      data-testid="currency_type"
                    >
                      ETH
                    </span>
                     
                    <span>
                      by 
                    </span>
                     
                    <span
                      data-testid="user"
                      title="0x658d22456633334234iF0cC"
                    >
                      0x658d22456633334...
                    </span>
                  </span>
                </h4>
                <div
                  class="ant-list-item-meta-description"
                >
                  <span
                    class="sc-eCImPb fYSFWD"
                    data-testid="date"
                  >
                    2021-11-17, 01:00 am
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li
            class="ant-list-item"
          >
            <div
              class="ant-list-item-meta"
            >
              <div
                class="ant-list-item-meta-avatar"
              >
                <div
                  class="sc-bdvvtL kGscza"
                >
                  <div
                    class="sc-gsDKAQ eNKYiV"
                  >
                     
                    <img
                      alt=""
                      class="sc-dkPtRN jsPQsZ"
                      height="28px"
                      src="/verified-account-avatar.svg"
                      width="26px"
                    />
                  </div>
                </div>
              </div>
              <div
                class="ant-list-item-meta-content"
              >
                <h4
                  class="ant-list-item-meta-title"
                >
                  <span
                    class="sc-jRQBWg jfyBYj"
                  >
                    <span
                      data-testid="crypto_value"
                    >
                      7.00698
                    </span>
                     
                    <span
                      class="sc-hKwDye emujjc"
                      data-testid="currency_type"
                    >
                      ETH
                    </span>
                     
                    <span>
                      by 
                    </span>
                     
                    <span
                      data-testid="user"
                      title="0x658d22456633334234iF0cC"
                    >
                      0x658d22456633334...
                    </span>
                  </span>
                </h4>
                <div
                  class="ant-list-item-meta-description"
                >
                  <span
                    class="sc-eCImPb fYSFWD"
                    data-testid="date"
                  >
                    2021-11-17, 01:00 am
                  </span>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
